presto-main/src/main/java/com/facebook/presto/operator/TaskContext.java

```diff
@@ -59,14 +61,19 @@
 {
     private final QueryContext queryContext;
     private final TaskStateMachine taskStateMachine;
+    private final GcMonitor gcMonitor;
     private final Executor notificationExecutor;
     private final ScheduledExecutorService yieldExecutor;
     private final Session session;
 
     private final long createNanos = System.nanoTime();
 
     private final AtomicLong startNanos = new AtomicLong();
+    private final AtomicLong startFullGcCount = new AtomicLong();
+    private final AtomicLong startFullGcTimeNanos = new AtomicLong();
     private final AtomicLong endNanos = new AtomicLong();
+    private final AtomicLong endFullGcCount = new AtomicLong();
+    private final AtomicLong endFullGcTimeNanos = new AtomicLong();
```

Comment:
These 3 fields likely deserve a comment about how they will (not just could) be set to a value smaller than start if a task is initially in a done state.

Reply:
I rewrote the code so that never happens.  I also cleaned up the regular start/end timing code in another commit